### PR TITLE
T3T1 vertical menu

### PR DESCRIPTION
**Pull request by obrusvit**
Vertical menu options. 
Finished for word select.
For tx context menu, some work will be done later.

### CODE REVIEW

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+#[cfg(feature = "ui_debug")]
+impl<T> crate::trace::Trace for VerticalMenu<T>
+where
+    T: AsRef<str>,
+{
+    fn trace(&self, t: &mut dyn crate::trace::Tracer) {
+        // TODO: implement
+        t.component("VerticalMenu");
+        // t.child("inner", &self.inner);
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs:152]: Could be something like
```suggestion
        t.in_list("buttons", |button_list| {
            for &button in self.buttons {
                button_list.child(button)
            }
        });
```

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            area: Rect::zero(),
+            buttons,
+            areas_sep: AreasForSeparators::new(),
+        }
+    }
+    pub fn select_word(words: [T; 3]) -> Self {
+        let mut buttons_vec = VerticalMenuButtons::new();
+        for (i, word) in words.into_iter().enumerate() {
+            let button = Button::with_text(word).styled(theme::button_vertical_menu());
+            buttons_vec.push(button);
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs:56]: Compiler complains that we're ignoring the return values of `push()`. Either use `unwrap!` or explicitly ignore them with `let _ = ...`.

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        None
+    }
+
+    fn paint(&mut self) {
+        // TODO remove when ui-t3t1 done
+    }
+
+    fn render<'s>(&'s self, target: &mut impl Renderer<'s>) {
+        // render buttons separated by thin bars
+        for button in self.buttons.iter() {
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs:127]: I think it's more ususal to do:
```suggestion
        for button in &self.buttons {
```

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    fn new(buttons: VerticalMenuButtons<T>) -> Self {
+        Self {
+            area: Rect::zero(),
+            buttons,
+            areas_sep: AreasForSeparators::new(),
+        }
+    }
+    pub fn select_word(words: [T; 3]) -> Self {
+        let mut buttons_vec = VerticalMenuButtons::new();
+        for (i, word) in words.into_iter().enumerate() {
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs:54]: Enumerate not needed?

File: core/embed/rust/src/ui/model_mercury/component/button.rs [hunk 5]
```diff
@@ -660,4 +622,48 @@ impl IconText {
...
     }
+    pub fn render<'s>(
+        & self,
+        target: &mut impl Renderer<'s>,
+        area: Rect,
+        style: &ButtonStyle,
+        baseline_offset: Offset,
+    ) {
+        let width = style.font.text_width(self.text.as_ref());
+        let height = style.font.text_height();
```
Comment (mmilata) [inline core/embed/rust/src/ui/model_mercury/component/button.rs:633]: Unused? Also bunch of other variables in the file.

**Review by mmilata — COMMENTED** [review]
Commented some nitpicks but overall looking good!

File: core/embed/rust/src/ui/model_mercury/layout.rs [hunk 6]
```diff
@@ -2033,6 +2076,11 @@ pub static mp_module_trezorui2: Module = obj_module! {
     ///    iterable must be of exact size. Returns index in range `0..3`."""
     Qstr::MP_QSTR_select_word => obj_fn_kw!(0, new_select_word).as_obj(),
 
+    /// def show_tx_context_menu() -> int:
```
[the remark this answers is not in the document]
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/layout.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            let button = Button::with_text(word).styled(theme::button_vertical_menu());
+            buttons_vec.push(button);
+        }
+        Self::new(buttons_vec)
+    }
+
+    pub fn context_menu(options: [T; 3], icons: [Icon; 3]) -> Self {
```
[the remark this answers is not in the document]
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+        self.area = bounds;
+        let mut remaining = bounds;
+        for i in 0..N_ITEMS {
+            let (area_button, new_remaining) = remaining.split_top(MENU_BUTTON_HEIGHT);
+            self.buttons[i].place(area_button);
+            remaining = new_remaining;
+            if i < N_SEPS {
+                let (area_sep, new_remaining) = remaining.split_top(MENU_SEP_HEIGHT);
+                self.areas_sep.push(area_sep);
```
[the remark this answers is not in the document]
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs (hunk 1 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs (hunk 2 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs (hunk 4 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/vertical_menu.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/component/button.rs (hunk 5 above)
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/component/button.rs]: https://github.com/trezor/trezor-firmware/pull/3599/commits/5664655e7aaf7b92db44c1e60bac06e9d3b56df0

File: core/embed/rust/src/ui/model_mercury/layout.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (mmilata) [reply core/embed/rust/src/ui/model_mercury/layout.rs]: Now it needs `make gen` again:(

**Review by mmilata — APPROVED** [review]
Thanks for the changes, please run `make gen`, squash into a single commit, and merge if there are no CI failures in addition to what `ui-t3t1` has (bunch of things fail due to drawlib branch having broken tests).

File: core/embed/rust/src/ui/model_mercury/layout.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (obrusvit) [reply core/embed/rust/src/ui/model_mercury/layout.rs]: Done in rebase

**Pull request merged**